generic3g/vertical/FixedLevelsVerticalGrid.F90

```diff
@@ -147,6 +147,44 @@ impure elemental logical function not_equal_FixedLevelsVerticalGrid(a, b) result
       not_equal = .not. (a==b)
    end function not_equal_FixedLevelsVerticalGrid
 
+   ! Create an ESMF_Field containing a 3D array that is replicated from
+   ! a 1D array at each point of the horizontal grid
+   function esmf_field_create_(geom, farray1d, rc) result(field)
+      type(ESMF_Field) :: field ! result
+      type(ESMF_Geom), intent(in) :: geom
+      real(kind=REAL32), intent(in) :: farray1d(:)
+!#      character(len=*), intent(in) :: vloc
+      integer, optional, intent(out) :: rc
+
+      integer, allocatable :: local_cell_count(:)
+      real(kind=REAL32), pointer :: farray3d(:, :, :)
```

Comment:
Should the ESMF_KIND be used here like this?
```suggestion
      real(kind=REAL32), pointer :: farray3d(:, :, :)
```

Reply:
Note - this code only shows up as mine because of a merge conflict.